owncloud/phoenix
# acceptance tests

I've played a bit with the nightwatch testing framework. Discussion about different frameworks in #79 
Here is a fist PoC with it.

How to run the tests:
1. download selenium (https://docs.seleniumhq.org/download/) and chrome webriver (https://sites.google.com/a/chromium.org/chromedriver/downloads)
1. unpack the webdriver and place the content in the same folder as the selinium jar file
1. start selenium with `java -jar selenium-server-standalone-x.x.x.jar -port 4445`
1. configure phoenix, (copy  https://github.com/owncloud/phoenix/blob/acceptance-tests-nightwatch/config.json.sample to `config.json` and edit it)
1. run phoenix `make run SERVER_HOST=0.0.0.0:8300`
1. run tests ` npm run acceptance-tests`

What should happen:
1. a browser window opens
1. selects the first ownCloud instance
1. types username and password
1. logs in and checks if the files list is dispayed

This is a very basic setup so far but I would like to hear your thoughts an comments
@felixheidecke @OshanIvantha @DeepDiver1975 @phil-davis @patrickjahns
